test/test.py

```diff
@@ -0,0 +1,98 @@
+          # all segments are more than 100, less than 500
+            self.assertTrue(len(x.split()) > 250)
+            self.assertTrue(len(x.split()) <= 502)
+          # all segments end on punctuation or are 500 words
+            self.assertTrue(
+                len(x.split()) == 500 or x[-1] == "." or x[-1] == "!" or x[-1] == "?")
+
+        for x in munge.split_corpus("test_files/quixote.txt"):
+          # all segments are more than 100, less than 500
+            self.assertTrue(len(x.split()) <= 500)
+            self.assertTrue(len(x.split()) > 250)
+          # all segments end on punctuation or are 500 words
+            self.assertTrue(
+                len(x.split()) == 500 or x[-1] == "." or x[-1] == "!" or x[-1] == "?")
+
+        # handles directories with text files and other kinds of files
+        for x in munge.split_corpus("test_files/"):
+          # all segments are more than 100, less than 500
+            self.assertTrue(len(x.split()) <= 500)
+            self.assertTrue(len(x.split()) > 250)
+          # all segments end on punctuation or are 500 words
+            self.assertTrue(
+                len(x.split()) == 500 or x[-1] == "." or x[-1] == "!" or x[-1] == "?")
+
+        # throws exn if source is not a txt file or directory ?? need to test exceptions?
+        # self.assertRaises(munge.split_corpus("dfd"),
+        #                   FileNotFoundError)
+        # self.assertRaises(munge.split_corpus("test_files/whale.txt"),
+        #                   FileNotFoundError)
+        # self.assertRaises(munge.split_corpus("test_files/fake.cry"),
+        #                   NotADirectoryError)
+
+    def test_write_clean_corpus(self):
+
+        # every line has correct formatting
+        text = munge.split_corpus("test_files/simple_whale_100.txt")
+        ids = list(range(len(text)))
+        names = ["w" + str(x) for x in ids]
+        munge.write_clean_corpus(
+            text, ids, names, "test_files/clean_whale_100.txt")
+        with open("test_files/clean_whale_100.txt", "r") as inf:
+            for i, line in enumerate(inf):
+                chunks = line.split("\t")
+                self.assertEqual(chunks[0], str(ids[i]))
+                self.assertEqual(chunks[1], names[i])
+                self.assertEqual(len(chunks), 3)
+
+        text = munge.split_corpus("test_files/quixote.txt")
+        ids = list(range(len(text)))
+        names = ["w" + str(x) for x in ids]
+        munge.write_clean_corpus(
+            text, ids, names, "test_files/clean_quixote.txt")
+        with open("test_files/clean_quixote.txt", "r") as inf:
+            for i, line in enumerate(inf):
+                chunks = line.split("\t")
+                self.assertEqual(chunks[0], str(ids[i]))
+                self.assertEqual(chunks[1], names[i])
+                self.assertEqual(len(chunks), 3)
+
+        # ids are unique (i could raise an error if not)
```

Comment:
I think it's good to check that IDs are unique. Note that you can use the assert* functions to do slightly more interesting things if you want by adding an additional argument, e.g.
```self.assertEqual(len(set(ids)), len(ids), "IDs are not unique!")```

Reply:
i was wanting to add this to the original function in munge.py rather than in tests, so that the user would be alerted. does that seem appropriate?